src/DynamoApplications/AssemblyConflictChecker.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using DynamoApplications.Properties;
+
+namespace Dynamo.Applications
+{
+    /// <summary>
+    /// Checks for incompatibilities between loaded assemblies.
+    /// </summary>
+    public class AssemblyConflictChecker : IDisposable
+    {
+        /// <summary>
+        /// List of assembly names to ingnore when checking for incompatibilities.
+        /// </summary>
+        public IEnumerable<string> AssemblyNamesToIgnore = new List<string>();
+
+        /// <summary>
+        ///  Flag specifying whether to skip all checks.
+        /// </summary>
+        public bool Enable;
+
+        /// <summary>
+        /// The default white list of dependencies to be ignored.
+        /// </summary>
+        private readonly string[] DefaultAssemblyNamesToIgnore = new string[] { "Newtonsoft.Json" };
+        public Exception[] Exceptions { get { return Conflicts.Where(x => x.Value == false).Select(x => x.Key).ToArray(); } }
+
+        private readonly Dictionary<Exception, bool> Conflicts = new Dictionary<Exception, bool>();
+
+        private void AssemblyLoad(object sender, AssemblyLoadEventArgs args)
+        {
+            if (!Enable)
+            {
+                return;
+            }
+            var conflicts = GetVersionMismatchedReferencesInAppDomain(args.LoadedAssembly);
+            var ignoreList = AssemblyNamesToIgnore?.Concat(DefaultAssemblyNamesToIgnore) ?? DefaultAssemblyNamesToIgnore;
+
+            foreach (var conflict in conflicts)
+            {
+                var ignored = ignoreList.Contains(conflict.Key);
+                Conflicts.Add(conflict.Value, ignored);
+            }
+        }
+
+        /// <summary>
+        /// Constructor for AssemblyConflictNotifier
+        /// </summary>
+        public AssemblyConflictChecker()
+        {
+            Enable = true;
+            AppDomain.CurrentDomain.AssemblyLoad += AssemblyLoad;
+        }
+
+        /// <summary>
+        /// Dispose function for AssemblyConflictNotifier
+        /// </summary>
+        public void Dispose()
+        {
+            AppDomain.CurrentDomain.AssemblyLoad -= AssemblyLoad;
+        }
+
+        /// <summary>
+        /// Logs any assembly conflicts found.
+        /// </summary>
+        public void LogConflicts(Models.DynamoModel DynamoModel)
+        {
+            foreach (var conflict in Conflicts)
+            {
+                bool isIgnored = conflict.Value;
+                var x = conflict.Key;
+
+                if (isIgnored)
+                {
+                    DynamoModel.Logger.LogInfo(GetType().ToString(), x.Message);
+                } else
+                {
+                    DynamoModel.Logger.LogNotification(DynamoModel.GetType().ToString(),
+                    x.GetType().ToString(), Resources.MismatchedAssemblyVersionShortMessage,
+                    x.Message);
+                }
```

Comment:
Should we log anything for ignored assemblies ? or should I make the message more simplistic instead of the whole exception message ?
Not sure there is a real difference between LogInfo and LogNotification.

Reply:
I think LogNotification bubbles up a notification in the UI and LogInfo does not.